bundles/framework/divmanazer/component/FileInput.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @class Oskari.userinterface.component.FileInput
+ *
+ */
+Oskari.clazz.define('Oskari.userinterface.component.FileInput', function (localization) {
+    this.loc = localization;
+    this.el = null;
+    this._template = {
+        fileBox: _.template('<div class="oskari-fileinput" style="display:none"> '+
+                    '<form method="post" action="", enctype="multipart/form-data" class="box" id="fileinput">'+
+                        '<div class="box__input">'+
+                            '<input type="file" name="file" id="file" class="box__file" />'+
+                            '<label>  <%= fileupload %> <label for="file" style="cursor: pointer;"> <a> <%= link %> </a> </label> </label> '+
+                        '</div>'+
+                        '<div class="box__uploading"> <%= uploading %>&hellip;</div>'+
+                        '<div class="box__success"><%= success %></div>'+
+                        '<div class="box__error"><%= error %></div>'+
+                    '</form>' +
+                '</div>')
+    }
+}, {
+        getElement: function () {
+            return this.el;
+        },
+        setElement: function ( el ) {
+            this.el = jQuery(el);
+        },
+        /**
+         * @method canUseAdvancedUpload
+         *
+         * Checks if the browser supports drag and drop events aswell as formdata & filereader
+         * @return {boolean} true if supported 
+         */
+        canUseAdvancedUpload: function() {
+            var div = document.createElement('div');
+            return (('draggable' in div) || ('ondragstart' in div && 'ondrop' in div)) && 'FormData' in window && 'FileReader' in window;
+         },
+                /**
+         * @method handleDragAndDrop
+         * Checks for drag and drop events, on submit makes ajax request
+         * 
+         *
+         */
+        handleDragAndDrop: function( cb ) {
+            var me = this;
+            var form = this.getElement().find('.box');
+            var input = form.find('input[type="file"]');
+            form.addClass('has-advanced-upload');
+            var droppedFiles = false;
+
+            form.on('drag dragstart dragend dragover dragenter dragleave drop', function(e) {
+                e.preventDefault();
+                e.stopPropagation();
+            })
+            .on('dragover dragenter', function() {
+                form.addClass('is-dragover');
+            })
+            .on('dragleave dragend drop', function() {
+                form.removeClass('is-dragover');
+            })
+            .on('drop', function(e) {
+                droppedFiles = e.originalEvent.dataTransfer.files;
```

Comment:
If this component only handles single file, should we check that the user only dropped one file?